assignments/lambda-classes.js

```diff
@@ -1 +1,104 @@
 // CODE here for your Lambda Classes
+
+class Person{
+    constructor(attr){
+        this.name = attr.name;
+        this.age = attr.age;
+        this.location = attr.location;
+    }
+    speak(){
+        console.log(`Hello my name is ${this.name}, I am from ${this.location}`)
+    }
+}
+
+const objectPerson = new Person({
+    name: 'Fred',
+    age: '25',
+    location: 'Bedrock'
+})
+
+objectPerson.speak();
+
+class Instructor extends Person{
+    constructor(attr){
+        super(attr);
+        this.specialty = attr.specialty;
+        this.favLanguage = attr.favLanguage;
+        this.catchPhrase = attr.catchPhrase;
+    }
+    demo(subject){
+        console.log(`Today we are learning about ${subject}`)
+
+    }
+    grade(student, subject){
+     student = {
+            name: 'Kofi'
+        }
+        subject = 'React'
+        
+        console.log(`${student.name} recieves a perfect score on ${subject}`)
+    }
+}
+
+
+
+const objectInstructor = new Instructor({
+    specialty: 'redux',
+    favLanguage: 'JavaScript',
+    catchPhrase: 'Don\'t forget the homies',
+    student: 'kofi'
+})
+
+objectInstructor.demo('closures');
+objectInstructor.grade();
+
+class Student extends Person{
+    constructor(attr){
+        super(attr);
+        this.previousBackground = attr.previousBackground;
+        this.className = attr.className;
+        this.favSubjects = attr.favSubjects;
+    }
+    listsSubjects(){
+        console.log(this.favSubjects);
+    }
+    PRAssignment(student, subject){
+        console.log(`${student.name} has submitted a PR for ${subject}`)
```

Comment:
The instructions say this method should have just one parameter, which is the subject. The name should come from the instance of the class because it is an extension of Person which has the **name** property. i.e

```
PRAssignment(subject){
        console.log(`${this.name} has submitted a PR for ${subject}`)
    }
```

So when you are creating an instance of the Class, add the name of the student there. For example,

```
const objectStudent = new Student({
    name: "Bolaji",
    previousBackground: 'computer science',
    className: 'CS132',
    favSubjects: ['HTML', 'CSS', 'JavaScript']
})
```